Sources/Sentry/SentrySdkInfo.m

```diff
@@ -29,16 +29,95 @@ typedef NS_ENUM(NSUInteger, SentryPackageManagerOption) {
 static SentryPackageManagerOption SENTRY_PACKAGE_INFO = SentryPackageManagerUnkown;
 #endif
 
+static NSSet<NSDictionary<NSString *, NSString *> *> *extraPackages;
+
 NS_ASSUME_NONNULL_BEGIN
 
 @interface SentrySdkInfo ()
-
-@property (nonatomic) SentryPackageManagerOption packageManager;
-
 @end
 
 @implementation SentrySdkInfo
 
++ (void)addPackageName:(NSString *)name version:(NSString *)version
+{
+    if (name == nil || version == nil) {
+        return;
+    }
+
+    @synchronized(extraPackages) {
+        NSDictionary<NSString *, NSString *> *newPackage =
+            @{ @"name" : name, @"version" : version };
+        if (extraPackages == nil) {
+            extraPackages = [[NSSet alloc] initWithObjects:newPackage, nil];
+        } else {
+            extraPackages = [extraPackages setByAddingObject:newPackage];
+        }
+    }
+}
+
++ (NSMutableSet<NSDictionary<NSString *, NSString *> *> *)getExtraPackages
+{
+    @synchronized(extraPackages) {
+        if (extraPackages == nil) {
+            return [[NSMutableSet alloc] init];
+        } else {
+            return [extraPackages mutableCopy];
+        }
+    }
+}
+
+#if TEST || TESTCI
++ (void)clearExtraPackages
+{
+    extraPackages = [[NSSet alloc] init];
+}
```

Comment:
`m`: It would seems a bit cleaner to me if we extract the extra packages code into its own class. We could even write it in Swift, similar to `SentryEnabledFeaturesBuilder`.

Reply:
We can't write it in Swift because the test suit (data comparisons) will break on the non-deterministing dict entries order after serialization.

Reply:
> non-deterministing dict entries order after serialization

Wait what? Can't we fix the data comparison? That sounds stupid 😢

Reply:
We can, but it doesn't make sense to block this PR by refactoring the tests.